src/main/java/core/basesyntax/WriteDataImpl.java

```diff
@@ -0,0 +1,33 @@
+package core.basesyntax;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+public class WriteDataImpl implements WriteData {
+
+    private String newPath;
+
+    public WriteDataImpl(String newPath) {
+        this.newPath = newPath;
+    }
+
+    @Override
+    public File writeData(List<String> fruitData) {
+        File file = new File(newPath);
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(file))) {
+            bufferedWriter.write("fruit,quantity");
+            bufferedWriter.newLine();
+
+            for (String fruit : fruitData) {
+                bufferedWriter.write(fruit); // Записуємо інформацію про фрукт
+                bufferedWriter.newLine(); // Додаємо новий рядок після кожного запису
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Error writing data to a  file",e);

```

Comment:
There is a typo in the exception message: "to a  file" should be "to a file". Also, consider including the file path in the message for better context: `"Error writing data to file: " + newPath`.